eland/tasks.py

```diff
@@ -185,6 +184,33 @@ def __repr__(self) -> str:
         return f"('{self._task_type}': ('sort_field': '{self._sort_field}', 'count': {self._count}))"
 
 
+class SampleTask(SizeTask):
+    def __init__(self, count):
+        super().__init__("sample")
+        self._count = count
+
+    def resolve_task(
+        self,
+        query_params: QUERY_PARAMS_TYPE,
+        post_processing: List["PostProcessingAction"],
+        query_compiler: "QueryCompiler",
+    ) -> RESOLVED_TASK_TYPE:
+        query_params["query"].random_score()
+
+        if query_params.get("query_size") is not None:
+            query_params["query_size"] = min(self._count, query_params["query_size"])
+        else:
+            query_params["query_size"] = self._count
+
```

Comment:
Something else to think about here is we want to order by `_score`. I actually think when I checked this out locally and added the `query_params["query_sort_order"] = "score"` I found a bug in `TailTask` not picking up the current `query_sort_order` when resolving tasks. Something to potentially investigate outside of this issue.